Source/JavaScriptCore/runtime/FunctionConstructor.cpp

```diff
@@ -146,6 +146,24 @@ JSObject* constructFunction(JSGlobalObject* globalObject, const ArgList& args, c
     auto code = stringifyFunction(globalObject, args, functionName, functionConstructionMode, scope, functionConstructorParametersEndPosition);
     EXCEPTION_ASSERT(!!scope.exception() == code.isNull());
 
+    if (Options::useTrustedTypes() && globalObject->requiresTrustedTypes()) {
+        bool isTrusted = true;
+        auto* structure = globalObject->trustedScriptStructure();
+        for (size_t i = 0; i < args.size(); i++) {
+            auto arg = args.at(i);
+
+            if (!arg.isObject() || structure != arg.structureOrNull()) {
+                isTrusted = false;
+                break;
+            }
+        }
+
+        if (!isTrusted && !globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::Function, code, args)) {
```

Comment:
This does not have exception check.

Reply:
Ditto.

Reply:
I've added a RETURN_IF_EXCEPTION inside of and after the if statement to catch both branches.

Reply:
That's not correct. Regardless of whether the exception is thrown, we need to check, so this code will gets crash with assertion.
You need to make it like,
```
if (!isTrusted) {
    bool canCompileStrings = globalObject->globalObjectMethodTable()->canCompileStrings(globalObject, CompilationType::Function, code, args);
    RETURN_IF_EXCEPTION(scope, { });
    ...
```

Please ensure that all of the added code is following to the above pattern precisely if it can throw an error / it is using exception scope.